syntax = "proto3";

package envoy.test.extensions.filters.http.ext_proc.v3;

option java_package = "io.envoyproxy.envoy.test.extensions.filters.http.ext_proc.v3";
option java_outer_classname = "TestProcessingRequestModifierProto";
option java_multiple_files = true;

// A test modifier config. Demonstrates a sample use case of a custom
// modifier that remaps attributes.
message TestProcessingRequestModifierConfig {
  // A map of request attributes to set in the attributes struct.
  // The key is the attribute name, the value is the attribute value,
  // interpretable by CEL. This allows for the re-mapping of attributes.
  map<string, string> mapped_request_attributes = 1;
}
